test/test_jit_cuda_fuser.py

```diff
@@ -721,7 +723,45 @@ def test_binary_ops(self):
                       torch.lt]
         for op, dtype in itertools.product(operations, data_types):
             if (dtype not in self.int_types) or (op not in skip_for_integer):
-                self._binary_test_helper(op, dtype)
+                self._binary_test_helper(op, dtype, True)  # random data
+                # disabled special numbers because of incorrect handling
+                # self._binary_test_helper(op, dtype, False) # special numbers
+
+    @unittest.skipIf(not RUN_CUDA, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    def test_binary_ops_type_promotion(self):
```

Comment:
Just curious what kind of errors were happening due to mixed data type. The might be more failing cases in mixed value types, value types meaning `N-Dim Tensor (N>0)`, `0-Dim Tensor`, `Scalar`.

The type promotion rule on eager mode treats inputs of different value types differently, e.g:
`Tensor + Scalar` and `3dim Tensor + 0-dim Tensor`. `test_category_rule` had some simple coverage on these scenarios.